contributors/design-proposals/raw-block-pv.md

```diff
@@ -592,23 +592,42 @@ Other considerations:
 ## New BlockVolumeMapper interface proposed design
 
 ```
- type BlockVolumeMapper interface {
-       Volume
-       CanBlockMap() error
-       SetUpDevice(podUID types.UID) error
-       SetUpDeviceAt(dir string, podUID types.UID) error
-       GetDevicePath() (string, error)
-       GetGlobalMapPath(spec *Spec) (string, error)
-       GetPodDeviceMapPath() string
- }
- 
- type BlockVolumeUnmapper interface {
-       Volume
-       TearDownDevice() error
-       TearDownDeviceAt(dir string) error
-       GetGlobalUnmapPath(spec *Spec) (string, error)
-       GetPodDeviceUnmapPath() string
- }
+type BlockVolumeMapper interface {
+	Volume
+	// SetUpDevice prepares the volume to a self-determined directory path,
+	// which may or may not exist yet and returns combination of physical
+	// device path of a block volume and error.
+	// If the plugin is non-attachable, it should prepare the device
+	// in /dev/ (or where appropriate) and return device path.
+	// If the plugin is attachable, it should not do anything, just return
+	// empty string for device path.
+	// This may be called more than once, so implementations must be idempotent.
+	SetUpDevice() (string, error)
+	// GetGlobalMapPath returns a global map path which contains
+	// a symbolic links associated to a block device.
+	// ex. plugins/kubernetes.io/{PluginName}/{DefaultKubeletVolumeDevicesDirName}/{volumeName}
```

Comment:
Is the only information we need from the plugin the unique volume name?  It seems like plugin name, and devices dir name should be known to the volume manager.  Can we just use the GetVolumeName call?

Reply:
@msau42 

Sorry for the confusion. The comment is wrong. The path contains
- volumePluginDependentPath for volume reconstruction
- pod's uuid

```
-	// ex. plugins/kubernetes.io/{PluginName}/{DefaultKubeletVolumeDevicesDirName}/{volumeName}
+	// ex. plugins/kubernetes.io/{PluginName}/{DefaultKubeletVolumeDevicesDirName}/{volumePluginDependentPath}/{pod uuid}
```

Reply:
@erinboyd Could you apply document update patch which I sent yesterday? Before @msau42 starts to review the interface change and mount point update, we need to apply that patch to avoid confusion.
Thanks.